src/indexed.js

```diff
@@ -0,0 +1,32 @@
+import {Seq} from 'immutable';
+import Base from './base';
+import {updateCursor} from './utils';
+
+function Indexed(root, data, keyPath, onChange, size) {
```

Comment:
I had pondered about using `class`, however, I can't see how the inheritance pattern I'm using, can fit elegantly into the model that the `class` syntax promotes. For example, in the case of `Indexed` and `Keyed` constructors, I'm inheriting methods from the prototype chains of both `Seq.Indexed` and `Seq.Keyed` respectively, whilst subsequently also 'inheriting' (merging) methods from `Base.prototype` for each. The issue is, `class` and `prototype` promote the differential inheritance - which this pattern isn't.